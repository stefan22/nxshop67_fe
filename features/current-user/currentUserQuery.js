import { gql } from '@apollo/client'
export const currentUserQuery = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        email
        name
      }
    }
  }
`
