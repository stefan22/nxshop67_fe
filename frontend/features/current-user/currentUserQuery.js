import { gql } from '@apollo/client'

const currentUserQuery = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        email
        name
      }
    }
  }
`

export default currentUserQuery
