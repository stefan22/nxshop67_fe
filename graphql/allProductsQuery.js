import { useQuery } from '@apollo/client'
import gql from 'graphql-tag'

const allProductsQuery = gql`
  query allProductsQuery {
    allProducts {
      id
      name
      price
      description
      photo {
        id
        image {
          publicUrlTransformed
        }
      }
    }
  }
`

const useAllProducts = () => {
  const {
    data: allProds,
    error: allProdsError,
    loading: allProdsLoading
  } = useQuery(allProductsQuery)

  return {
    allProds,
    allProdsError,
    allProdsLoading
  }
}

export default useAllProducts
