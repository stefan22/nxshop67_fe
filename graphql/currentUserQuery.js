import { gql, useQuery } from '@apollo/client'

export const currentUserQuery = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        email
        name
      }
    }
  }
`

const useCurrentUser = () => {
  const {
    data: currentUser,
    loading: userLoading,
    error: userError
  } = useQuery(currentUserQuery)

  if (userLoading) return <p>Loading...</p>
  if (userError) return console.log(userError)

  return {
    currentUser: currentUser?.authenticatedItem
  }
}

export { useCurrentUser }
