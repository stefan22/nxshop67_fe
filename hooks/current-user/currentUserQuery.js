import { gql, useQuery } from '@apollo/client'

export const currentUserQuery = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        email
        name
      }
    }
  }
`
