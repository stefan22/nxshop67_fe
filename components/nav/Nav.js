import React from 'react'
import Link from 'next/link'
import NavContainer from './Nav.styles'
import SignOut from '../SignOut'

import { useCurrentUser } from '../../hooks/current-user'

const Nav = () => {
  const { currentUser } = useCurrentUser()

  return (
    <NavContainer data-testid="menu-links">
      <Link href="./products">Products</Link>
      {currentUser && (
        <>
          <Link href="./account">Account</Link>
          <Link href="./order">Order</Link>
          <SignOut />
        </>
      )}
      {!currentUser && <Link href="./signin">Signin</Link>}
    </NavContainer>
  )
}

export default Nav
