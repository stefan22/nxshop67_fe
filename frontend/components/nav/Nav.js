import React from 'react'
import Link from 'next/link'
import NavContainer from './Nav.styles'
import SignOut from '../signout'

import { useCurrentUser } from '../../features/current-user'

const Nav = () => {
  const { user, loading } = useCurrentUser()

  if (loading) return <p>Loading....</p>

  return (
    <NavContainer data-testid="menu-links">
      <Link href="/products">Products</Link>
      {user && (
        <>
          <Link href="/account">Account</Link>
          <SignOut />
        </>
      )}
      <Link href="/contact-us">Contact-us</Link>
      {!user && <Link href="/signin">Signin</Link>}
    </NavContainer>
  )
}

export default Nav
