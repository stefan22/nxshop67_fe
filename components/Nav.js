import React from 'react'
import Link from 'next/link'
import NavSl from './styles/NavSl'
import { useCurrentUser } from '@/graphql/currentUserQuery'

const Nav = () => {
  const { currentUser } = useCurrentUser()

  return (
    <NavSl data-testid="menu-links">
      <Link href="/products">Products</Link>
      {currentUser && (
        <>
          <Link href="/account">Account</Link>
          <Link href="/order">Order</Link>
          <Link href="/signout">Signout</Link>
        </>
      )}
      {!currentUser && <Link href="/signin">Signin</Link>}
    </NavSl>
  )
}

export default Nav
