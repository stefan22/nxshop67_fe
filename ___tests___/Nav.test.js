/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { MockedProvider } from "@apollo/react-testing";
import Nav from '../components/Nav'
import { currentUserQuery } from '../graphql/currentUserQuery'

const getUser = () => ({
  id: 12345,
  name: 'Al Capone',
  email: 'capone@gmail.com'
})

const notSignedIn = [
  {
    request: { query: currentUserQuery },
    result: { data: { authenticatedItem: null }},
  }
]

const signedIn = [
  {
    request: { query: currentUserQuery },
    result: { data: { authenticatedItem: getUser()}}
  }
]


describe('Nav menu', () => {

  it('nav menu when user not signed in', () => {
      const { container, debug } = render(
        <MockedProvider mocks={notSignedIn}>
          <Nav />
        </MockedProvider>
      )

      const menuLinks = screen.getByTestId('menu-links');
      const signin = screen.getByText('Signin');
      const products = screen.getByText('Products')

      expect(menuLinks).toBeInTheDocument()
      expect(container).toHaveTextContent('Products')
      expect(container).toHaveTextContent('Signin')
      expect(signin).toHaveAttribute('href', '/signin')
      expect(products).toHaveAttribute('href','/products')
      expect(container).not.toHaveTextContent('Signout')
      expect(container).not.toHaveTextContent('Account')
      expect(container).toMatchSnapshot()
  })

  it('nav menu when user is signed in', async() => {
    const { container, debug } = render(
      <MockedProvider mocks={signedIn}>
        <Nav />
      </MockedProvider>
    )
    const account = await screen.findByText('Account')
    const signout = await screen.findByText('Sign Out')

    expect(container).toHaveTextContent('Account')
    expect(account).toHaveAttribute('href', '/account')
    expect(account).not.toHaveTextContent('href','/order')
    expect(container).not.toHaveTextContent('Signin')
    expect(container).toHaveTextContent('Sign Out')
    expect(signout).not.toHaveAttribute('href','/signout')
    expect(container).toMatchSnapshot();


  })

})
