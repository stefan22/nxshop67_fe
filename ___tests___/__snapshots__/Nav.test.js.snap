// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav menu nav menu when user is signed in 1`] = `
<div>
  <ul
    class="NavSl-sc-gqsck-0 eWpGRM"
    data-testid="menu-links"
  >
    <a
      href="/products"
    >
      Products
    </a>
    <a
      href="/account"
    >
      Account
    </a>
    <a
      href="/order"
    >
      Order
    </a>
    <button
      type="button"
    >
      Sign Out
    </button>
  </ul>
</div>
`;

exports[`Nav menu nav menu when user not signed in 1`] = `
<div>
  <ul
    class="NavSl-sc-gqsck-0 eWpGRM"
    data-testid="menu-links"
  >
    <a
      href="/products"
    >
      Products
    </a>
    <a
      href="/signin"
    >
      Signin
    </a>
  </ul>
</div>
`;
